`hash_extender` with OpenSSL 3.0

I have compiled `hash_extender` on Ubuntu 20.04 with OpenSSL 1.1 and when trying to run the resulting binary on Ubuntu 22.04 (with OpenSSL 3.0) this is the error:

> hash_extender: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory

Trying to build it again results in a lot of warnings (treated eventually as errors):

```bash
$ make
[CC] buffer.o
[CC] formats.o
[CC] hash_extender.o
[CC] hash_extender_engine.o
hash_extender_engine.c: In function ‘md4_hash’:
hash_extender_engine.c:340:3: error: ‘MD4_Init’ is deprecated: Since OpenSSL 3.0                                                                                                                                                                               [-Werror=deprecated-declarations]
  340 |   MD4_Init(&c);
      |   ^~~~~~~~
In file included from hash_extender_engine.c:26:
/usr/include/openssl/md4.h:50:27: note: declared here
   50 | OSSL_DEPRECATEDIN_3_0 int MD4_Init(MD4_CTX *c);
      |                           ^~~~~~~~
[...]
cc1: all warnings being treated as errors
make: *** [Makefile:39: hash_extender_engine.o] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ OS		:= $(shell uname | tr '/[[:lower:]]' '_[[:upper:]]')
 
 # These are the specifications of the toolchain
 CC		:= gcc
-CFLAGS		:= -std=c89 -g -oS -Wall -Werror -Wno-deprecated
+CFLAGS		:= -std=c89 -g -oS -Wall -Wno-deprecated
 CPPFLAGS	:= -D_DEFAULT_SOURCE -D$(OS) $(WHIRLPOOL)
 ifeq ($(OS),DARWIN)
     LDFLAGS		:= -lssl -lcrypto -L/usr/local/opt/openssl/lib/

